aten/src/ATen/Context.h

```diff
@@ -485,7 +485,8 @@ inline DeprecatedTypeProperties& MPS(ScalarType s) {
 }
 
 inline bool hasCUDA() {
-  return globalContext().hasCUDA();
+  return globalContext().hasCUDA() &&
```

Comment:
Could you leave the same comment as in the PR description in this file saying the compile time vs runtime versions?

Reply:
I leave the comments in this file. Is it OK?

Reply:
Isn't this a BC breaking change? Or there were no `hasCUDA` in 2.5?

Reply:
Hi @malfet  For `at::hasCUDA`, it is not a BC-breaking change. However, for `globalContext().hasCUDA()`, it could be considered a BC-breaking change.
We plan to modify `globalContext().hasCUDA()` to check for CUDA availability at compile time (based on the build) instead of runtime. Do you have any concerns about this change?